extensions/security/deployment/src/test/java/io/quarkus/security/test/permissionsallowed/checker/PermissionCheckerActionsNotSupportedValidationFailureTest.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.security.test.permissionsallowed.checker;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import jakarta.inject.Singleton;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.security.PermissionChecker;
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class PermissionCheckerActionsNotSupportedValidationFailureTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .assertException(t -> {
+                var exceptionMessage = t.getMessage();
+                assertEquals(IllegalArgumentException.class, t.getClass(), exceptionMessage);
+                assertTrue(t.getMessage().contains("SecuredBean#check"), exceptionMessage);
+                assertTrue(t.getMessage().contains("actions are currently not supported"), exceptionMessage);
+            });
+
+    @Test
+    public void test() {
+        Assertions.fail();
+    }
+
+    @Singleton
+    public static class SecuredBean {
+
+        @PermissionsAllowed("read:all")
+        public void securedBean() {
+            // EMPTY
+        }
+
+        @PermissionChecker("read:all")
```

Comment:
Michal, the problem is, `:` convention where we map anything after `:` as Java Permission actions can be difficult to understand to users who do not create custom Java Permissions.
`:` is a very common separator, it is everywhere like in https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes.

So when I, as a user, write this code and get `the actions are not supported` error, it makes me confused.
I'm sorry, I don't understand. But in any case, we can continue in 2025, thanks

Reply:
This is well documented both in the https://github.com/quarkusio/quarkus-security/blob/main/src/main/java/io/quarkus/security/PermissionsAllowed.java and https://quarkus.io/guides/security-authorize-web-endpoints-reference#permission-annotation. If users don't read docs, I don't know what to do. Also, I still don't believe what you mention is https://github.com/quarkusio/quarkus/issues/45224.